source/README.adoc

```diff
@@ -0,0 +1,142 @@
+
+:toc:
+:toc-placement!:
+= RabbitMQ Source
+
+The RabbitMQ source translates messages on a RabbitMQ exchange to CloudEvents
+over HTTP for use with Knative Eventing. The Source can bind to an existing
+exchange or create a new one, as needed.
+
+toc::[]
+
+
+== Published Events
+
+All messages received by the Source will be published with the following schema:
+
+.Table Event Attributes
+|===
+| Attribute | Value | Notes
+
+| `type` | `dev.knative.rabbitmq.event` |
+| `source` | `/apis/v1/namespace/*$NS*/rabbitmqsources/*$NAME*#*$TOPIC*`
+   | `NS`, `NAME` and `TOPIC` are derived from the Source configuration.
+| `id` | a unique id | This uses the MessageId if available, and a UUID otherwise.
+| `subject` | the message's id | empty string if no message ID is present.
+| `datacontenttype` | `application/json` | Currently static.
+| `key` | the message's id | empty string if no message ID is present.
+|===
+
+The event's payload will be set to the data content of the message.
+
+=== Samples
+
+(in CloudEvents JSON format):
```

Comment:
Since there's if/else in the payload (for example, subject is empty string if no message ID is present), I wonder if we should also include a RabbitMQ message as input and then showing how it gets transformed into a CloudEvent.

Reply:
Interestingly, it seems like the ability to publish a rabbitmq event from the command line is harder than I'd expect. I found a `rabbitmqadmin` here that can do it: https://www.rabbitmq.com/management-cli.html